ssh.go

```diff
@@ -83,7 +83,7 @@ func (t *sshSmartSubtransport) Action(urlString string, action SmartServiceActio
 			}
 			t.Close()
 		}
-		cmd = fmt.Sprintf("git-upload-pack %q", u.Path)
+		cmd = fmt.Sprintf("git-upload-pack '%s'", u.Path)
```

Comment:
can `\` and `'` be escaped? otherwise one of these characters can be added to the path and cause some unwanted shenanigans on the remote host.

also, the link that was shared made me realize that we are currently missing the %-decoding that's done in libgit2! https://github.com/libgit2/libgit2/blob/358a60e1b46000ea99ef10b4dd709e92f75ff74b/src/str.c#L455-L481  since this does not regress on that regard, it's fine if that is not fixed in this same commit (to keep its scope small), _but_ would it be possible to add a small `TODO` for us to fix it later?

Reply:
Thanks for pointing that out.
I've tried escaping them. Please correct me if it's not right.
And also, if the TODO message is reasonable.